vectortile/src/main/scala/geotrellis/vectortile/Command.scala

```diff
@@ -0,0 +1,143 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.vectortile
+
+// --- //
+
+/** VectorTile geometries are stored as packed lists of "Command Integers".
+  * There are currently three legal Commands: MoveTo, LineTo, and ClosePath.
+  * Each adhere to the following format:
+  *
+  * [ ... 00000 | 000 ]
+  * -------------------
+  *       |        |
+  *       |        --- The "Command ID". MoveTo: 001 (1)
+  *       |                              LineTo: 010 (2)
+  *       |                              ClosePath: 111 (7)
+  *       |
+  *       --- The remaining 29 bits are the "Parameter Count".
+  *           This indicates the number of _pairs_ of ints
+  *           that follow that should be interpreted as Z-encoded
+  *           deltas from the current "cursor".
+  *
+  * The "cursor" is the location of the current vertex being considered,
+  * and it starts at (0,0) for each Feature. As MoveTo and LineTo commands
+  * are read, the cursor moves according the list of parsed delta pairs.
+  *
+  * Caveats:
+  * - Point features, whether single or multi, will always consist of a single MoveTo.
+  * - Any Polygon in a Polygon feature must have a LineTo with a count of at least 2.
+  * - ClosePath must always have a parameter count of 1.
+  */
+sealed trait Command
+
+case class MoveTo(deltas: Array[(Int,Int)]) extends Command
+
+case class LineTo(deltas: Array[(Int,Int)]) extends Command
+
+case object ClosePath extends Command
+
+case class InvalidCommand(id: Int, count: Int) extends Exception
+
+object Command {
+  /** Attempt to parse a list of Command/Parameter Integers. */
+  // TODO (+:) is bad
+  def commands(cmds: Array[Int]): Array[Command] = cmds match {
+    case Array() => Array()
+    case ns => parseCmd(ns.head) match {
+      case (1,count) => {
+        val (ps,rest) = ns.tail.splitAt(count * 2)
+        val res = new Array[(Int,Int)](count)
+        var i = 0
+
+        while(i < count) {
+          res.update(i, (unzig(ps(i*2)), unzig(ps(i*2+1))))
+
+          i += 1
+        }
+
+        MoveTo(res) +: commands(rest)
+      }
+      case (2,count) => {
+        val (ps,rest) = ns.tail.splitAt(count * 2)
+        val res = new Array[(Int,Int)](count)
+        var i = 0
+
+        while(i < count) {
+          res.update(i, (unzig(ps(i*2)), unzig(ps(i*2+1))))
+
+          i += 1
+        }
+
+        LineTo(res) +: commands(rest)
+      }
+      case (7,_) => ClosePath +: commands(ns.tail)
+    }
+  }
```

Comment:
@lossyrob `+:` seems to copy the source array, which probably murders performance. Recursion was also the most intuitive here, but is probably also bad.